### PR TITLE
BB2-409: Migrate build/deploy Jenkins job to CloudBees Core

### PR DESCRIPTION
**Pull request by njdister**
**JIRA Ticket:**

[BB2-409](https://jira.cms.gov/browse/BB2-409)

**User Story or Bug Summary:**

Migrate the following jobs to CB Core:
- DEPLOY - Static Site

### What Does This PR Do?

- Defines and automates the build of a docker container suitable static site build and deploy
- Defines a Kubernetes pod for use in CB Core
- Rewrite the build and deploy job for simplification and parameterization per environment, just like our AMI deploys.  No more "all environments at once" job.
    - This adds extra time to each build because gem/bundle components aren't shared between environments, but I believe this is worth the trade-off in terms of simplicity in logic.

### What Should Reviewers Watch For?

If you're reviewing this PR, please check these things, in particular:

- Did I break anything that used to work?
- Did I miss porting over some vital feature?
- Like the Deploy AMI jobs, I have opted not to have the environment placeholders.  Is there any need for that?

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then here's a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence: N/A.
    * Does this PR add any new software dependencies? **No**.
    * Does this PR modify or invalidate any of our security controls? **No**.
    * Does this PR store or transmit data that was not stored or transmitted before? **No**.
* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR should not be merged unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons? **No**.

### What Needs to Be Merged and Deployed Before this PR?

There are no prerequisites, but some postrequisites once this PR is merged are:
- Disable all jobs in CBJ
- Switch all jobs in CBC to use the master branch

### Submitter Checklist

I have gone through and verified that...:

* [x] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [x] I have named this PR and its branch such that they'll be automatically be linked to the (most) relevant Jira issue, per: <https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html>.
* [ ] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [x] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [x] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. BFD, SLS, etc.).
* [x] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: ops/Jenkinsfile.cbc-build-deploy [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+      steps {
+        script {
+          echo "Setting SITE_BUCKET from SITE_ENV"
+          if (params.SITE_ENV == "test") {
+            SITE_BUCKET = "test.bluebutton.cms.gov"
+          } else if (params.SITE_ENV == "staging") {
+            SITE_BUCKET = "staging.bluebutton.cms.gov"
+          } else if (params.SITE_ENV == "prod") {
+            SITE_BUCKET = "bluebutton.cms.gov"
+          }
```
Comment (dtisza1) [inline ops/Jenkinsfile.cbc-build-deploy:49]: @njdister Should there be a final `ELSE` block that produces an error, if `params.SITE_ENV` has an invalid value?

**Review by dtisza1 — APPROVED** [review]
This looks good to me!  

Just one review comment that might not be needed.

File: ops/Jenkinsfile.cbc-build-deploy (hunk 1 above)
Comment (njdister) [reply ops/Jenkinsfile.cbc-build-deploy:49]: Yep, thats easy to add just in case.  👍

**Review by jzulim — APPROVED** [review]
LGTM. Thanks for the demo and walkthrough. Nice job on simplifying the parameters and the pipeline(s) overall.

**Review by dtisza1 — APPROVED** [review]
Looks great!  TY for adding that :-)

**Pull request merged**